lib/xenops_server.ml

```diff
@@ -2144,6 +2155,19 @@ module PCI = struct
       (fun () ->
          debug "PCI.list %s" vm;
          DB.list vm) ()
+
+  let dequarantine _ dbg (pci_addr:Pci.address) =
+    let module B = (val get_backend () : S) in
+    let f () =
+      debug "PCI.dequarantine %s" B.PCI.(string_of_address pci_addr);
+      match B.PCI.dequarantine pci_addr with
```

Comment:
I'm not sure whether the best place for this is here or inside B.PCI.dequarantine itself. The atom above would bypass this check and call B.PCI.dequarantine directly.
The safest is probably to implement it the way it is currently written, writing it the other way around could introduce problems when applying Xen as an update depending whether you have the hotfix introducing quarantining or not?

Reply:
I think raising the error in the backend is probably best.